daemon/cmd/daemon.go

```diff
@@ -218,6 +219,9 @@ type Daemon struct {
 	settings cellSettings
 	// enable modules health support
 	healthProvider cell.Health
+
+	// Tunnel-related configuration
+	tunnelConfig tunnel.Config
```

Comment:
Sorry for my newbie question.

The DI container instantiates a tunnel.Config and injects it here automatically so that features relying on ad-hoc tunnel such as DSR GENEVE and Egress GW can retrieve the tunnel config as Single Source of Truth? Is that this PR aiming at?

Reply:
Yes, essentially it gets injected into any provide and invoke function that declares it as parameter (either directly or through a struct with the cell.In field). More specifically, in this specific case it gets injected [here](https://github.com/cilium/cilium/blob/7a6ed1452e404f2e9af75d0d4dae552b802e357c/daemon/cmd/daemon_main.go#L1681), and then just copied into the Daemon struct. This allows to have a single source of truth about whether e.g., the tunnel device needs to be created and its configuration, without depending on any global variable/function.

Leveraging the same DI strategy, this PR also allows additional modules (e.g., egress gateway/DSR-geneve) to specifically request that the tunnel is created even if Cilium is configured in native routing mode. But it also guarantees that any consumer of `tunnel.Config` will always see the final materialized configuration (thanks to the DAG constructed by hive/dig for the initialization order).